sync/src/block/extension.rs

```diff
@@ -663,6 +663,16 @@ impl NetworkExtension<Event> for Extension {
                     channel.send(*peer).unwrap();
                 }
             }
+            Event::GetPeerBestBlockHashes(channel) => {
+                for header_downloader in self.header_downloaders.values() {
+                    channel.send(header_downloader.best_hash()).unwrap();
```

Comment:
We need the information which peer has which best block hash.
Please pass the IP address of the peer also.